### PR TITLE
group renewals by policy (not one transaction per item)

### PR DESCRIPTION
**Pull request by Schparky**

### CODE REVIEW

File: application/actions/ledger.go [hunk 1]
```diff
@@ -163,7 +163,11 @@ func ledgerAnnualProcess(c buffalo.Context) error {
 
 	currentYear := time.Now().UTC().Year()
 
-	if err := models.ProcessAnnualCoverage(tx, currentYear); err != nil {
+	var policies models.Policies
+	if err := policies.AllActive(tx); err != nil {
+		return err
```
Comment (Baggerone) [inline application/actions/ledger.go:168]: Should this be `return reportError(c, err)`?
(If so, it looks like there might be a couple of other places in the file that may need to change.)

File: application/models/policy.go [hunk 2]
```diff
@@ -528,3 +532,64 @@ func (p *Policy) createInvite(tx *pop.Connection, invite api.PolicyUserInviteCre
 	}
 	return nil
 }
+
+// ProcessAnnualCoverage creates coverage renewal ledger entries for all items covered for the given year.
+// Does not create new records for items already processed.
+func (p *Policies) ProcessAnnualCoverage(tx *pop.Connection, year int) error {
+	for _, pp := range *p {
+		if err := pp.ProcessAnnualCoverage(tx, year); err != nil {
+			return fmt.Errorf("error processing annual coverage for policy %s: %w", pp.ID, err)
```
Comment (Baggerone) [inline application/models/policy.go:541]: I'm wondering about the mix of returning normal errors and AppErrors (lines 553 and 560).
Are you OK with that?

**Review by Baggerone — APPROVED** [review]
A couple of things to consider.

File: application/actions/ledger.go (hunk 1 above)
Comment (Schparky) [reply application/actions/ledger.go]: Thanks for spotting these

File: application/models/policy.go (hunk 2 above)
Comment (Schparky) [reply application/models/policy.go:541]: Thanks. Might have been ok, but much better to be consistent.

**Pull request merged**